File: test-junit5/src/test/java/io/micronaut/test/junit5/DisableResolveParametersTest.java
package io.micronaut.test.junit5;

import io.micronaut.context.annotation.Property;
import io.micronaut.context.annotation.Requires;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import jakarta.inject.Singleton;
import java.util.stream.Stream;

@Property(name = "spec.name", value = "ResolveParametersTest")
@MicronautTest(resolveParameters = false)
class DisableResolveParametersTest {
    static Stream<Arguments> fooArgs() {
        return Stream.of(Arguments.of(new Foo()));
    }

    @ParameterizedTest
    @MethodSource("fooArgs")
    void foo(Foo arg) { // <1>
        Assertions.assertNotNull(arg);
    }

    @Test
    @Disabled("Doesn't work with resolverParameters set to false") // <2>
    void bar(Foo arg) {
        Assertions.assertNotNull(arg);
    }

    @Requires(property = "spec.name", value = "ResolveParametersTest")
    @Singleton
    static class Foo {
    }
}

File: test-suite-python/src/test/python/micronaut/test/python/DisableResolveParametersTest.py
from jakarta.inject import Singleton
from micronaut.context.annotation import Property, Requires
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Disabled, Test


@Requires(property="spec.name", value="DisableResolveParametersTest")
@Singleton
class Foo:
    pass


@Property(name="spec.name", value="DisableResolveParametersTest")
@MicronautTest(resolveParameters=False)
class DisableResolveParametersTest:

    @Test
    def foo(self):  # <1>
        pass

    @Test
    @Disabled("Doesn't work with resolveParameters set to False")  # <2>
    def bar(self, arg: Foo):
        assert arg is not None
